promql/engine_test.go

```diff
@@ -814,6 +956,16 @@ load 10s
 			Start:      time.Unix(0, 0),
 			End:        time.Unix(10, 0),
 			Interval:   5 * time.Second,
+		}, {
+			// The step of evaluation time aligned subqueries divides evenly into the range, so:
+			//   - The subquery takes 20 samples now, with 10 for each bigmetric.
+			//   - Hence, the buffer is now 10 samples.
+			//   - We still have the last 2, so a total of 32.
```

Comment:
Shouldn't this work out exactly as with the conventional alignment, as everything is aligned the same?

As said above, things get interesting when the evaluation timestamp is not a multiple of the step anyway.

Reply:
Yes, when removing `+ newEv.interval` it's the same.